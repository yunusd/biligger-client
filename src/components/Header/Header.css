#header-menu-fix {
  background-color: red;
  margin-bottom: 100px;
}

.header-menu {
  background-color: #fff !important;
  border-bottom:none!important;
  box-shadow: none!important;
}


.header-right {
  padding:0!important;
  color:gray!important;
}

.header-right:last-of-type{
  padding-right:.92857143em!important;
}