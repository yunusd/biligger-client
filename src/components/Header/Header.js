import React from 'react';
import { Link } from 'react-router-dom';
import { ApolloConsumer } from 'react-apollo';
import axios from 'axios';

import {
  Menu, Image, Button, Dropdown,
} from 'semantic-ui-react';
import './Header.css';
import logo from '../../logo.png';

import LogIn from './LogIn';
import { Category } from '../Category';
import GET_AUTH_STATUS from '../../queries';

const AppHeader = (props) => {
  const handleSubmit = (client) => {
    axios.get('https://localhost:3000/auth/logout', { withCredentials: 'include' })
      .then(() => { client.clearStore(); window.location.replace('/'); })
      .catch(error => error);
  };
  if (props.location.pathname !== '/giriş' && props.location.pathname !== '/kayıt') {
    return (
      <div id="header-menu-fix">
        <Menu fixed="top" inverted className="header-menu">
          <Menu.Item header className="borderless">
            <Image as={Link} to="/" src={logo} width="100px" />
          </Menu.Item>
          <Menu.Item>
            <Category />
          </Menu.Item>
          <ApolloConsumer>
            {(client) => {
              const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });

              return (
                currentUser.isLoggedIn
                  ? (
                    <Menu.Item position="right" className="header-right">
                      <Link to="/yeni-bilig"><Button color="green">Yeni Bilig</Button></Link>
                      &nbsp;
                      &nbsp;
                      &nbsp;
                      <Dropdown button text="Profil">
                        <Dropdown.Menu>
                          <Dropdown.Item>Ayarlar</Dropdown.Item>
                          <Dropdown.Item onClick={() => { handleSubmit(client); }}>
                            Çıkış
                          </Dropdown.Item>
                        </Dropdown.Menu>
                      </Dropdown>
                    </Menu.Item>
                  )
                  : (
                    <Menu.Item position="right" className="header-right">
                      <LogIn button={<Button basic color="green">Giriş Yap</Button>} />
                    </Menu.Item>
                ));
            }}
          </ApolloConsumer>
        </Menu>
      </div>
    );
  }
  return <div />;
};

export default AppHeader;
