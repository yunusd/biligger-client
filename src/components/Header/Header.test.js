import React from 'react';
import { shallow } from 'enzyme';
import {Â Dropdown } from 'semantic-ui-react';

import Header from './Header';

describe('header', () => {
  it('render without crashing', () => {
    const wrapper = shallow(<Header />).exists();
    expect(wrapper).toBeTruthy();
  });

  it('render correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('if user login return different menu', () => {
    localStorage.setItem('access_token', 'dummy-token');
    const wrapper = shallow(<Header />);
    const html = wrapper.find(Dropdown).html();
    expect(html).toContain('Profil');
  });
});
