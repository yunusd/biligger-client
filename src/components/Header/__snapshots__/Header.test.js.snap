// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Menu
        fixed="top"
        inverted={true}
      >
        <MenuItem
          className="borderless"
          header={true}
        >
          <Link
            replace={false}
            to="/"
          >
            <Image
              as="img"
              src="logo.png"
              ui={true}
              width="100px"
            />
          </Link>
        </MenuItem>
        <MenuItem>
          <Category />
        </MenuItem>
        <MenuItem
          className="header-right"
          position="right"
        >
          <LogIn
            button={
              <Button
                as="button"
                basic={true}
              >
                Giriş Yap
              </Button>
            }
          />
              yada    
          <Register
            button={
              <Button
                as="button"
                basic={true}
                color="green"
              >
                Kayıt Ol
              </Button>
            }
          />
        </MenuItem>
      </Menu>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            className="borderless"
            header={true}
          >
            <Link
              replace={false}
              to="/"
            >
              <Image
                as="img"
                src="logo.png"
                ui={true}
                width="100px"
              />
            </Link>
          </MenuItem>,
          <MenuItem>
            <Category />
          </MenuItem>,
          <MenuItem
            className="header-right"
            position="right"
          >
            <LogIn
              button={
                <Button
                  as="button"
                  basic={true}
                >
                  Giriş Yap
                </Button>
              }
            />
                yada    
            <Register
              button={
                <Button
                  as="button"
                  basic={true}
                  color="green"
                >
                  Kayıt Ol
                </Button>
              }
            />
          </MenuItem>,
        ],
        "fixed": "top",
        "inverted": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              <Image
                as="img"
                src="logo.png"
                ui={true}
                width="100px"
              />
            </Link>,
            "className": "borderless",
            "header": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                as="img"
                src="logo.png"
                ui={true}
                width="100px"
              />,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "img",
                "src": "logo.png",
                "ui": true,
                "width": "100px",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Category />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <LogIn
                button={
                  <Button
                    as="button"
                    basic={true}
                  >
                    Giriş Yap
                  </Button>
                }
              />,
              "    yada    ",
              <Register
                button={
                  <Button
                    as="button"
                    basic={true}
                    color="green"
                  >
                    Kayıt Ol
                  </Button>
                }
              />,
            ],
            "className": "header-right",
            "position": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": <Button
                  as="button"
                  basic={true}
                >
                  Giriş Yap
                </Button>,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "    yada    ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "button": <Button
                  as="button"
                  basic={true}
                  color="green"
                >
                  Kayıt Ol
                </Button>,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Menu
          fixed="top"
          inverted={true}
        >
          <MenuItem
            className="borderless"
            header={true}
          >
            <Link
              replace={false}
              to="/"
            >
              <Image
                as="img"
                src="logo.png"
                ui={true}
                width="100px"
              />
            </Link>
          </MenuItem>
          <MenuItem>
            <Category />
          </MenuItem>
          <MenuItem
            className="header-right"
            position="right"
          >
            <LogIn
              button={
                <Button
                  as="button"
                  basic={true}
                >
                  Giriş Yap
                </Button>
              }
            />
                yada    
            <Register
              button={
                <Button
                  as="button"
                  basic={true}
                  color="green"
                >
                  Kayıt Ol
                </Button>
              }
            />
          </MenuItem>
        </Menu>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              className="borderless"
              header={true}
            >
              <Link
                replace={false}
                to="/"
              >
                <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                  width="100px"
                />
              </Link>
            </MenuItem>,
            <MenuItem>
              <Category />
            </MenuItem>,
            <MenuItem
              className="header-right"
              position="right"
            >
              <LogIn
                button={
                  <Button
                    as="button"
                    basic={true}
                  >
                    Giriş Yap
                  </Button>
                }
              />
                  yada    
              <Register
                button={
                  <Button
                    as="button"
                    basic={true}
                    color="green"
                  >
                    Kayıt Ol
                  </Button>
                }
              />
            </MenuItem>,
          ],
          "fixed": "top",
          "inverted": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                  width="100px"
                />
              </Link>,
              "className": "borderless",
              "header": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  as="img"
                  src="logo.png"
                  ui={true}
                  width="100px"
                />,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "img",
                  "src": "logo.png",
                  "ui": true,
                  "width": "100px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Category />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <LogIn
                  button={
                    <Button
                      as="button"
                      basic={true}
                    >
                      Giriş Yap
                    </Button>
                  }
                />,
                "    yada    ",
                <Register
                  button={
                    <Button
                      as="button"
                      basic={true}
                      color="green"
                    >
                      Kayıt Ol
                    </Button>
                  }
                />,
              ],
              "className": "header-right",
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": <Button
                    as="button"
                    basic={true}
                  >
                    Giriş Yap
                  </Button>,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "    yada    ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "button": <Button
                    as="button"
                    basic={true}
                    color="green"
                  >
                    Kayıt Ol
                  </Button>,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
